src/mjolnir/pbfgraphparser.cc

```diff
@@ -2326,30 +2324,31 @@ struct graph_callback : public OSMPBF::Callback {
 
     // Transform tags. If no results that means the way does not have tags
     // suitable for use in routing.
-    Tags results = tags.empty() ? empty_way_results_ : lua_.Transform(OSMType::kWay, osmid_, tags);
-    if (results.empty()) {
+    const Tags tags =
+        way.tags().empty() ? empty_way_tags_ : lua_.Transform(OSMType::kWay, osmid_, way.tags());
+    if (tags.empty()) {
```

Comment:
I'm not sure about this `Tags results` -> `Tags tags` renaming as it obscures this PR. At the same time, IMO it makes it much easier to read everything below.
Please let me know if you think that this renaming should be reverted.